app/src/main/cpp/HandMeshRenderer.cpp

```diff
@@ -99,6 +105,97 @@ void HandMeshRendererSpheres::Update(const uint32_t aControllerIndex, const std:
         handMesh.sphereTransforms[i]->SetTransform(handJointTransforms[i]);
 }
 
+// HandMeshRendererGeometry
+
+struct HandMeshGeometry {
+    vrb::TogglePtr toggle;
+    vrb::GeometryDrawablePtr geometry;
+};
+
+struct HandMeshRendererGeometry::State {
+    std::vector<HandMeshGeometry> handMeshState;
+};
+
+HandMeshRendererPtr HandMeshRendererGeometry::Create(vrb::CreationContextPtr& aContext) {
+    return std::make_unique<vrb::ConcreteClass<HandMeshRendererGeometry, HandMeshRendererGeometry::State> >(aContext);
+}
+
+HandMeshRendererGeometry::HandMeshRendererGeometry(State& aState, vrb::CreationContextPtr& aContext)
+        : m(aState) {
+    context = aContext;
+}
+
+void HandMeshRendererGeometry::Initialize(HandMeshGeometry& handMesh, const vrb::GroupPtr& aRoot) {
+    vrb::CreationContextPtr create = context.lock();
+    handMesh.toggle = vrb::Toggle::Create(create);
+    aRoot->AddNode(handMesh.toggle);
+
+    auto geom = vrb::GeometryDrawable::Create(create);
+
+    handMesh.toggle->AddNode(geom);
+
+    auto renderBuffer = vrb::RenderBuffer::Create(create);
+    renderBuffer->DefinePosition(offsetof(HandMeshVertexMSFT, position), 3);
+    renderBuffer->DefineNormal(offsetof(HandMeshVertexMSFT, normal), 3);
+    geom->SetRenderBuffer(renderBuffer);
+
+    auto state = GetHandMeshDefaultRenderState(create);
+    geom->SetRenderState(state);
+
+    handMesh.geometry = std::move(geom);
+}
+
+void HandMeshRendererGeometry::Update(const uint32_t aControllerIndex, const std::vector<vrb::Matrix>& handJointTransforms,
+                                      const vrb::GroupPtr& aRoot, const bool aEnabled, const bool leftHanded) {
+    if (aControllerIndex >= m.handMeshState.size())
+        m.handMeshState.resize(aControllerIndex + 1);
+    auto& handMesh = m.handMeshState.at(aControllerIndex);
+
+    // We need to call ToggleAll() even if aEnabled is false, to be able to hide the
+    // hand mesh.
+    if (handMesh.toggle)
+        handMesh.toggle->ToggleAll(aEnabled);
+
+    if (!aEnabled)
+        return;
+
+    // Lazily initialize toggle, transform and geometry nodes, and render buffers
+    if (!handMesh.toggle)
+        Initialize(handMesh, aRoot);
+}
+
+void HandMeshRendererGeometry::Update(const uint32_t aControllerIndex, const std::vector<vrb::Matrix>& handJointTransforms,
+                                      const vrb::GroupPtr& aRoot, HandMeshBufferPtr& aBuffer, const bool aEnabled, const bool leftHanded) {
+    Update(aControllerIndex, handJointTransforms, aRoot, aEnabled, leftHanded);
+
+    if (!aBuffer || !aEnabled)
```

Comment:
It was not there because it is part of the other version of Update(), which I call right above.

This is not relevant anymore since I removed the other version of Update(), so updating the toggle now happens here.